### PR TITLE
Avoid doxygen warnings

### PR DESCRIPTION
**Pull request by masterleinad**
Together with #9876 this fixes all `doxygen` warnings for me.
- `EXTRACT_PRIV_VIRTUAL` doesn't seem to exist in my `doxygen` version but the default value is `OFF` anyway.
- I couldn't quite get rid of the last warnings without disabling `GENERATE_DEPRECATEDLIST`. Personally, I don't find that page really useful, but I am, of course, open for other opinions.
- The same goes for the include grpahs

### CODE REVIEW

File: doc/doxygen/options.dox.in [hunk 1]
```diff
@@ -175,22 +174,37 @@ PREDEFINED             = DOXYGEN=1 \
                          DEBUG=1 \
                          DEAL_II_WARNING(x)= \
                          DEAL_II_USE_MT_POSIX=1 \
+                         DEAL_II_CONSTEXPR=constexpr \
+                         DEAL_II_NAMESPACE_OPEN= \
```
Comment (tjhei) [inline doc/doxygen/options.dox.in:178]: did you reorder these define with a specific ordering in mind? Maybe we should sort alpabetically?

File: doc/doxygen/options.dox.in [hunk 2]
```diff
@@ -202,31 +216,30 @@ PREDEFINED             = DOXYGEN=1 \
...
                          DEAL_II_WITH_SCALAPACK=1 \
                          DEAL_II_WITH_SLEPC=1 \
                          DEAL_II_WITH_SUNDIALS=1 \
+                         DEAL_II_SUNDIALS_WITH_IDAS=1 \
+                         DEAL_II_SUNDIALS_VERSION_GTE=1 \
+                         DEAL_II_SUNDIALS_VERSION_LT=0 \
                          DEAL_II_WITH_SYMENGINE=1 \
-                         DEAL_II_SYMENGINE_WITH_LLVM=1 \
-                         DEAL_II_WITH_THREADS=1 \
+                         DEAL_II_SYMENGINE_WITH_LLVM =1 \
```
Comment (tjhei) [inline doc/doxygen/options.dox.in:231]:
```suggestion
                         DEAL_II_SYMENGINE_WITH_LLVM=1 \
```

File: doc/doxygen/options.dox.in [hunk 3]
```diff
@@ -202,31 +216,30 @@ PREDEFINED             = DOXYGEN=1 \
...
-                         DEAL_II_DISABLE_EXTRA_DIAGNOSTICS= \
-                         DEAL_II_DEPRECATED= \
-                         DEAL_II_CUDA_HOST_DEV= \
-                         DEAL_II_ALWAYS_INLINE= \
-                         __device__= \
-                         DEAL_II_P4EST_VERSION_GTE=1 \
-                         DEAL_II_SUNDIALS_VERSION_GTE=1 \
-                         DEAL_II_SUNDIALS_VERSION_LT=0 \
-                         DEAL_II_TRILINOS_VERSION_GTE=1
+                         DEAL_II_WITH_ZLIB
```
Comment (tjhei) [inline doc/doxygen/options.dox.in:242]: why did you change this from ``=1``?

File: doc/doxygen/options.dox.in [hunk 4]
```diff
@@ -258,8 +271,8 @@ COLLABORATION_GRAPH    = NO
 GROUP_GRAPHS           = YES
 UML_LOOK               = NO
 TEMPLATE_RELATIONS     = YES
-INCLUDE_GRAPH          = YES
-INCLUDED_BY_GRAPH      = YES
+INCLUDE_GRAPH          = NO
```
Comment (tjhei) [inline doc/doxygen/options.dox.in:274]: I don't feel strongly about this change, but it is worth pointing out.

File: doc/doxygen/options.dox.in [hunk 5]
```diff
@@ -52,7 +51,7 @@ SORT_MEMBER_DOCS       = NO
 SORT_BRIEF_DOCS        = NO
 SORT_BY_SCOPE_NAME     = NO
 GENERATE_TODOLIST      = YES
-GENERATE_DEPRECATEDLIST= YES
+GENERATE_DEPRECATEDLIST= NO
```
Comment (tjhei) [inline doc/doxygen/options.dox.in:54]: not particularly useful, so I am okay.

File: doc/doxygen/options.dox.in (hunk 1 above)
Comment (masterleinad) [reply doc/doxygen/options.dox.in:178]: Yes, it's `DEAL_WITH_*` options (and corresponding configurations) last, everything else before that. I extracted the former ones from `base/config.h.in` and would rather not merge them with the others.

File: doc/doxygen/options.dox.in (hunk 3 above)
Comment (masterleinad) [reply doc/doxygen/options.dox.in:242]: That wasn't intentional... Thanks!

**Comment by masterleinad** [comment]
Instead of using #9876, I fixed (or rather silenced) the remaining complaints in `source` as well. I don't get any `doxygen` warnings with this pull request alone anymore.

**Comment by masterleinad** [comment]
For clarification: disabling the include graphs is related to these warnings:
```
warning: Included by graph for 'config.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'exceptions.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'fe.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'memory.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'utilities.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'memory_consumption.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'mpi.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'point.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'quadrature_lib.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'smartpointer.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'subscriptor.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'tensor.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'tria.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'dof_accessor.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'dof_handler.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'fe_values.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'tria_iterator.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'block_vector.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
warning: Included by graph for 'vector.h' not generated, too many nodes. Consider increasing DOT_GRAPH_MAX_NODES.
```
As I said before, I don't find great value in having these graphs (e.g. https://www.dealii.org/developer/doxygen/deal.II/rtree_8h.html), but an alternative solution would be to allow a larger number of nodes.

**Comment by tjhei** [comment]
> I don't find great value in having these graphs

I agree. It probably speeds up running ``doxygen`` as well. ;-)

**Comment by jppelteret** [comment]
> It probably speeds up running doxygen as well

Pretty much **anything** done to speed up doxygen gets my upvote!

**Comment by bangerth** [comment]
I can live without these graphs -- I think they have limited value.

File: doc/doxygen/options.dox.in [hunk 6]
```diff
@@ -202,31 +216,30 @@ PREDEFINED             = DOXYGEN=1 \
...
-                         DEAL_II_DISABLE_EXTRA_DIAGNOSTICS= \
-                         DEAL_II_DEPRECATED= \
-                         DEAL_II_CUDA_HOST_DEV= \
-                         DEAL_II_ALWAYS_INLINE= \
-                         __device__= \
-                         DEAL_II_P4EST_VERSION_GTE=1 \
-                         DEAL_II_SUNDIALS_VERSION_GTE=1 \
-                         DEAL_II_SUNDIALS_VERSION_LT=0 \
-                         DEAL_II_TRILINOS_VERSION_GTE=1
+                         DEAL_II_WITH_ZLIB=1
```
Comment (bangerth) [inline doc/doxygen/options.dox.in:242]: wow, damn, this list is already pretty long!

File: examples/step-20/step-20.cc [hunk 7]
```diff
@@ -755,7 +755,7 @@ namespace Step20
   // The last interesting function is the one in which we generate graphical
   // output. Note that all velocity components get the same solution name
   // "u". Together with using
-  // DataComponentInterpretation::::component_is_part_of_vector this will
+  // DataComponentInterpretation::component_is_part_of_vector this will
```
Comment (bangerth) [inline examples/step-20/step-20.cc:758]: :-)

File: include/deal.II/differentiation/sd/symengine_scalar_operations.h [hunk 8]
```diff
@@ -170,7 +170,7 @@ namespace Differentiation
      * valid symbols or symbolic expressions.
      *
      * @tparam ignore_invalid_symbols See the
-     * @ref add_to_symbol_map(types::substitution_map &, const Expression &)
+     * `add_to_symbol_map(types::substitution_map &, const Expression &)`
```
Comment (jppelteret) [inline include/deal.II/differentiation/sd/symengine_scalar_operations.h:173]: Do these still link back to the referenced function?

File: examples/step-69/step-69.cc [hunk 9]
```diff
@@ -1314,7 +1314,7 @@ namespace Step69
     // $\mathbf{m} \cdot \boldsymbol{\nu}_i =0$ on the entirety of the
     // boundary we should preserve the density and total (mechanical)
     // energy. This requires us to modify the $\mathbf{c}_{ij}$ vectors at
-    // the boundary as follows @cite GuermondEtAl2018:
+    // the boundary as follows @cite GuermondEtAl2018 :
```
Comment (bangerth) [inline examples/step-69/step-69.cc:1317]: the perils of operating off a poorly defined syntax...

File: include/deal.II/distributed/cell_data_transfer.h [hunk 10]
```diff
@@ -186,7 +192,7 @@ namespace parallel
         }
 
         /**
-         * @copydoc dealii::CoarseningStrategies::mean
+         * @copydoc dealii::CoarseningStrategies::mean()
```
Comment (bangerth) [inline include/deal.II/distributed/cell_data_transfer.h:195]: oh, that helps? I had stared at this for a few minutes and couldn't figure out what doxygen was missing...

**Comment by bangerth** [comment]
Let's wait for @masterleinad 's answer to @jppelteret 's lone question before merging.

File: include/deal.II/differentiation/sd/symengine_scalar_operations.h (hunk 8 above)
Comment (masterleinad) [reply include/deal.II/differentiation/sd/symengine_scalar_operations.h:173]: Yes, I still see the links.

**Comment by bangerth** [comment]
Question answered, let's merge!

File: include/deal.II/differentiation/sd/symengine_scalar_operations.h (hunk 8 above)
Comment (jppelteret) [reply include/deal.II/differentiation/sd/symengine_scalar_operations.h:173]: Awesome. Thanks a lot for the confirmation!
